### PR TITLE
Patches to the build system to properly support importing ioda-converters

### PR DESCRIPTION
**Pull request by BenjaminRuston**
this is a port of the changes in: https://github.com/JCSDA-internal/ioda-converters/pull/1549

PR1549 will be closed as a duplicate

## Description

This PR makes a few small fixes to the ioda-converters build system to allow it to be found using `find_package(iodaconv)` by dependent packages. I've fixed a few typos, and `make install` now works correctly. 

## Issue(s) addressed

None noted in JCSDA-internal, but this fixes some bugs on my end. Now I can reference `bufr2ioda.x` and related apps in my CMake projects.

I've made `ioda` a required package. CMake already threw cryptic errors when it wasn't found, and it doesn't really make sense to build ioda-converters without ioda anymore. Too many programs, scripts, and libraries depend on it.

## Dependencies

None

## Impact

Mildly positive.

## Checklist

- [x] I have performed a self-review of my own code
    - The changes work on my end.
- [x] I have made corresponding changes to the documentation
- [x] I have run the unit tests before creating the PR
    - The unit tests cannot test this capability, but I've tested manually with a project that runs `find_package(iodaconv REQUIRED)` and references `bufr2ioda.x`. It works.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -80,7 +82,7 @@ endif()
 find_package( gsl-lite HINTS $ENV{gsl_lite_DIR} )
 find_package( MPI )
 
-find_package( ioda QUIET )
+find_package( ioda REQUIRED QUIET )
```
Comment (climbfuji) [inline CMakeLists.txt]: I would remove QUIET if it is REQUIRED, but that's a personal choice

File: CMakeLists.txt [hunk 2]
```diff
@@ -9,15 +9,17 @@
 
 cmake_minimum_required( VERSION 3.12 )
 
-find_package( ecbuild 3 REQUIRED HINTS ${CMAKE_CURENT_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR}/../ecbuild )
 
 project( iodaconv VERSION 0.0.1 LANGUAGES CXX Fortran )
+list( APPEND CMAKE_MODULE_PATH ${CMAKE_CURRENT_SOURCE_DIR}/cmake )
 
+find_package( ecbuild 3 REQUIRED HINTS ${CMAKE_CURENT_SOURCE_DIR} ${CMAKE_CURRENT_SOURCE_DIR}/../ecbuild )
```
Comment (climbfuji) [inline CMakeLists.txt]: Why would anyone still build ecbuild as part of the code? Isn't that always coming from spack-stack?

Comment (BenjaminRuston) [reply CMakeLists.txt]: @rhoneyager-tomorrow ?

Comment (rhoneyager-tomorrow) [reply CMakeLists.txt]: The HINTS are taken from https://github.com/JCSDA/jedi-bundle/blob/develop/CMakeLists.txt#L6. Feel free to remove if you think it's unnecessary, but @BenjaminRuston now owns this PR and I can't make changes.

Comment (BenjaminRuston) [reply CMakeLists.txt]: think you can make a "suggestion" though @rhoneyager-tomorrow and may even be able to commit from there?

I'm happy to remove ,, also agree with the removal of `QUIET` Dom mentioned above

File: CMakeLists.txt (hunk 1 above)
Comment (BenjaminRuston) [inline CMakeLists.txt]:
```suggestion
find_package( ioda REQUIRED )
```

File: CMakeLists.txt (hunk 2 above)
Comment (BenjaminRuston) [inline CMakeLists.txt]:
```suggestion
```

Comment (BenjaminRuston) [reply CMakeLists.txt]: do not have a strong opinion on this one, @fmahebert could I ask your opinion?

please go ahead and commit suggestion if you think it is not necessary, fyi,, I'm leaning towards removal

Comment (rhoneyager-tomorrow) [inline CMakeLists.txt]:
```suggestion
find_package( ecbuild 3 REQUIRED )
```

Comment (rhoneyager-tomorrow) [reply CMakeLists.txt]: You do need to keep the `find_package( ecbuild ...` part. The HINTS are what could be removed.

Comment (BenjaminRuston) [reply CMakeLists.txt]: that makes a lot of sense ;0)

**Review by climbfuji — APPROVED** [review]

**Review by rhoneyager-tomorrow — APPROVED** [review]

**Review by fabiolrdiniz — APPROVED** [review]
Thanks, @rhoneyager-tomorrow & @BenjaminRuston!

**Review by srherbener — APPROVED** [review]

File: CMakeLists.txt (hunk 2 above)
Comment (fabiolrdiniz) [reply CMakeLists.txt]: Done in https://github.com/JCSDA-internal/ioda-converters/pull/1566/commits/89ff8650c21b508f57ec5a42463d3778cb05eca3.

**Pull request merged**